### PR TITLE
Hexagonal Number

### PR DESCRIPTION
**Pull request by tk315**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Article on Hexagonal Number

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#354

### CODE REVIEW

**Comment by tk315** [comment]
Please review sir @HarshCasper @iamrajiv @ankitaggarwal23 @atarax665 @gat786 @nidh-ish

**Comment by tk315** [comment]
Image has been modified for night mode

**Comment by tk315** [comment]
@HarshCasper please review

File: docs/mathematics/hexagonal-number.mdx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+slug: /mathematics/hexagonal-number
+---
+
+## Introduction
+
+A Hexagonal number is a figurate number. A Figurate number is a number that can be represented by a regular geometrical arrangement of equally spaced points. The `nth` hexagonal number `Hn` refers to the count of distinct dots which can be used to draw the outlines of regular hexagonal pattern such that hexagons are overlaid to share one vertex. We are given a number `n` and we need to print the `nth` hexagonal number.
+
+## Explanation 
+
+Every hexagonal number is a triangular number and every odd triangular number is a hexagonal number. For calculating the `nth` hexagonal number just apply the formula given below. First few hexagonal numbers are `1`, `6`, `15`, `28`, `45`, `66`, `91`, `120`, `153`, `190`.
```
Comment (iamrajiv) [inline docs/mathematics/hexagonal-number.mdx:12]:
```suggestion
Every hexagonal number is a triangular number and every odd triangular number is a hexagonal number. For calculating the `nth` hexagonal number just apply the formula given below. The first few hexagonal few hexagonal numbers are `1`, `6`, `15`, `28`, `45`, `66`, `91`, `120`, `153`, `190`.
```

File: docs/mathematics/hexagonal-number.mdx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+nth Hexagonal Number: n * (2 * n - 1) 
+```
+
+Test for Hexagonal Number:
+```
+We can test whether a positive integer `x` is a hexagonal number or not by calculating below expression:
+
+n = (sqrt(8*x + 1) + 1) / 4
+
+If n is integer then x is a hexagonal number
```
Comment (iamrajiv) [inline docs/mathematics/hexagonal-number.mdx:27]:
```suggestion
If n is an integer then x is a hexagonal number
```

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@tk315 Implement the suggested changes.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**